addons/sale/models/product_document.py

```diff
@@ -8,15 +8,22 @@ class ProductDocument(models.Model):
 
     attached_on = fields.Selection(
         selection=[
-            ('quotation', "Quotation"),
-            ('sale_order', "Confirmed order"),
+            ('hidden', "Hidden"),
+            ('quotation', "On quote"),
+            ('sale_order', "On confirm order"),
         ],
-        string="Visible at",
+        required=True,
+        default='hidden',
```

Comment:
This will need an upgrade script =] (I'm surprised upgrade is green btw :thinking: )

Reply:
`ondelete` is setting the empty fields as hidden
When I added this field to my existing database with some of the old documents it did indeed require the update but after adding the `ondelete` it no longer required the update
Let me know if its enough or if I need to write an upgrade script

Reply:
No need for ondelete, it's only (supposed to be) useful when you add new values with `selection_add`.
You'll indeed have to make an upgrade script to change empty values to 'hidden' so that the required constraint can be applied to the database.

Reply:
Yes, but you only have to define it in `sale_pdf_quote_builder`, for the additional `inside` value, so that when the module is uninstalled, it knows what it should do with records having the value `inside` as `attached_on`.